plugin/src/main/groovy/com/novoda/staticanalysis/internal/CollectViolationsTask.groovy

```diff
@@ -2,10 +2,12 @@ package com.novoda.staticanalysis.internal
 
 import com.novoda.staticanalysis.Violations
 import org.gradle.api.DefaultTask
+import org.gradle.api.tasks.InputFile
 import org.gradle.api.tasks.TaskAction
 
 abstract class CollectViolationsTask extends DefaultTask {
 
+    @InputFile
```

Comment:
Added this to make the task better cacheable. Don't think it will hurt in any way. WDYT?